tapdb/sqlc/queries/transfers.sql

```diff
@@ -42,20 +42,25 @@ SET proof_delivery_complete = @delivery_complete
 WHERE output_id = (SELECT output_id FROM target);
 
 -- name: QueryAssetTransfers :many
-SELECT
-    id, height_hint, txns.txid, transfer_time_unix
+SELECT DISTINCT
```

Comment:
I'm just saying that whenever I see a `DISTINCT` in a query, from experience, it's a sign that most likely something is fishy with the query. We're joining over an additional table, even though we don't need values from that table, so it should probably only be used in a sub-query (or maybe a CTE would fit here as well).
Also, I could imagine `DISTINCT` not being super fast, but that's more of a gut feeling.

Reply:
Also, I think we should have a basic unit tests covering the new query. I myself made multiple mistakes in queries just because I didn't add proper unit tests.

Reply:
Thinking about this a bit more, perhaps we want both `DISTINCT` and a better join? Here's my reasoning:

without `DISTINCT` we face two errors: duplicates and missing rows. But with `DISTINCT` we have to worry about missing rows. In other words, we do actually want the final rows to be `DISTINCT`.

As for performance, I think we need to optimise against an actual problem, and I don't think we've seen a performance problem just yet. And it's not clear to me that performance is actually significantly affected.

But I also see your point that the rest of the query has to remove duplicates also. Otherwise we might lose precision in what we want the statement to achieve.

Do you think using `DISTINCT` will hide problems though?

Reply:
I also don't understand what the `DISTINCT` is doing here - how would we end up with duplicate or missing rows for this query?

Reply:
Yeah, I'm pretty sure we don't need the `DISTINCT` anymore with the sub query. Before it was needed to de-duplicate because we selected/joined transfers **and** outputs but were only interested in fields of the transfers, so duplicates were possible.
But now we only select over transfers (and chain transactions, which should be a 1:1 relationship). So there shouldn't be any duplicates.
And I'm not sure what you mean by "missing rows"? Missing from what table?

Reply:
> And I'm not sure what you mean by "missing rows"? Missing from what table?

I mean that one error type is that the query doesn't return some rows that should be there. And that `DISTINCT` guarantees the absence of another error type: duplicate rows. It's nice to have that guarantee.

I agree that we don't **need** `DISTINCT` but we might as well have it because it is what we want. We want the returned rows from the query to be distinct.

Anyway, I'm going to remove `DISTINCT` for now because I don't think it matters very much. The JOIN has been modified as you suggested.